plugins/rest/aws.go

```diff
@@ -317,11 +317,11 @@ func (cs *awsWebIdentityCredentialService) refreshFromService() error {
 		"Version":          []string{"2011-06-15"},
 	}
 	stsRequestURL, _ := url.Parse(cs.stsPath())
-	stsRequestURL.RawQuery = queryVals.Encode()
 
 	// construct an HTTP client with a reasonably short timeout
 	client := &http.Client{Timeout: time.Second * 10}
-	req, err := http.NewRequest(http.MethodGet, stsRequestURL.String(), nil)
+	req, err := http.NewRequest(http.MethodPost, stsRequestURL.String(), strings.NewReader(queryVals.Encode()))
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
 	if err != nil {
```

Comment:
Can we do that after the err check? The chances are slim, but better safe than sorry...

Reply:
Ah yep, totally!